test/e2e/framework/pv_util.go

```diff
@@ -295,53 +354,72 @@ func CreatePVsPVCs(numpvs, numpvcs int, c clientset.Interface, ns string, pvConf
 
 	// create pvs and pvcs
 	for i = 0; i < pvsToCreate; i++ {
-		pv, pvc = CreatePVPVC(c, pvConfig, pvcConfig, ns, false)
+		if pv, pvc, err = CreatePVPVC(c, pvConfig, pvcConfig, ns, false); err != nil {
+			return pvMap, pvcMap, err
```

Comment:
Failing in the middle leaves the maps populated with however many successfully created objects there were at the time of an error.  In either an error or success case, the maps are returned with all existing resources.  From there, they are either consumed by the test or cleaned up in the `AfterEach`.  This reduces the need for duplicated code within the function scope (e.g. checking for error, deleting resources, returning, failing, attempting to delete resources again.)